src/OpenApi/Generate.elm

```diff
@@ -1208,43 +1418,232 @@ operationToSuccessTypeAndExpectAndResolver operation =
+                                                , toExpect = expectJsonBetter errorDecoders_ (Elm.val ("decode" ++ typeName))
+                                                , resolver = jsonResolverCustom.call errorDecoders_ <| Elm.val ("decode" ++ typeName)
+                                                }
                                             )
+                                            errorDecoders
+                                            errorTypeDeclaration
                                 )
             )
 
 
-jsonResolver :
+customHttpError : Elm.Declaration
+customHttpError =
+    Elm.customType "Error"
+        [ Elm.variantWith "BadUrl" [ Elm.Annotation.string ]
+        , Elm.variant "Timeout"
+        , Elm.variant "NetworkError"
+        , Elm.variantWith "KnownBadStatus" [ Elm.Annotation.int, Elm.Annotation.var "err" ]
+        , Elm.variantWith "UnknownBadStatus" [ Gen.Http.annotation_.metadata, Elm.Annotation.var "body" ]
+        , Elm.variantWith "BadErrorBody" [ Gen.Http.annotation_.metadata, Elm.Annotation.var "body" ]
+        , Elm.variantWith "BadBody" [ Gen.Http.annotation_.metadata, Elm.Annotation.var "body" ]
+        ]
+        |> Elm.exposeWith
+            { exposeConstructor = True
+            , group = Just "Types"
+            }
+
+
+expectJsonCustom :
     { declaration : Elm.Declaration
-    , call : Elm.Expression -> Elm.Expression
-    , callFrom : List String -> Elm.Expression -> Elm.Expression
+    , call : Elm.Expression -> Elm.Expression -> Elm.Expression -> Elm.Expression
+    , callFrom : List String -> Elm.Expression -> Elm.Expression -> Elm.Expression -> Elm.Expression
+    , value : List String -> Elm.Expression
+    }
+expectJsonCustom =
+    Elm.Declare.fn3 "expectJsonCustom"
+        ( "toMsg"
+        , Just
+            (Elm.Annotation.function
+                [ Gen.Result.annotation_.result
+                    (Elm.Annotation.namedWith [] "Error" [ Elm.Annotation.var "err", Elm.Annotation.string ])
+                    (Elm.Annotation.var "success")
+                ]
+                (Elm.Annotation.var "msg")
+            )
+        )
+        ( "errorDecoders"
+        , Just
+            (Gen.Dict.annotation_.dict
+                Gen.String.annotation_.string
+                (Gen.Json.Decode.annotation_.decoder (Elm.Annotation.var "err"))
+            )
+        )
+        ( "successDecoder"
+        , Just
+            (Gen.Json.Decode.annotation_.decoder (Elm.Annotation.var "success"))
+        )
+        (\toMsg errorDecoders successDecoder ->
+            Gen.Http.expectStringResponse (\result -> Elm.apply toMsg [ result ]) <|
+                \response ->
+                    -- NOTE: The below results in unexpected codegen like:
+                    --
+                    --     Http.BadUrl_ string.String ->
+                    --         Result.Err (BadUrl string.String)
```

Comment:
I _think_ this is fixed in the last elm-codegen

Reply:
I think so as well, but I'm stuck on an old version because elm-pages depends on an old version :(

Reply:
@dillonkearns :eyes:

Reply:
Let's so how the CI feels about bumping the dependency 😄 https://github.com/dillonkearns/elm-pages/pull/434

Reply:
Yeah, Cypress has been acting up on my builds lately 😢  I think it's working though because there are >0 passing cypress tests, which means it's running.

Reply:
I went ahead and merged. I'll sort out the cypress CI issues and do a deploy soon.

Reply:
GitHub was having issues recently, not sure if it'd impact your CI at all

Reply:
It's been happening intermittently for a while, so probably unrelated.

Reply:
Should be all good here now with the latest release, by the way!